astropy/io/votable/converters.py

```diff
@@ -424,14 +444,24 @@ def __init__(self, field, config=None, pos=None):
             self.binoutput = self._binoutput_var
             self.arraysize = "*"
         else:
-            field.arraysize = field.arraysize.removesuffix("*")
+            is_variable = field.arraysize.endswith("*")
+
+            # Get numeric part without modifying field.arraysize
+            numeric_part = field.arraysize.removesuffix("*")
```

Comment:
We could uniformize the comments here with between L326:L330 as currently they are complimenter for the same code content.

Reply:
Yes good point, should be done in new commit